cwms_radar_api/src/main/java/cwms/radar/data/dao/LocationsDaoImpl.java

```diff
@@ -257,43 +265,91 @@ public Catalog getLocationCatalog(String cursor, int pageSize, String unitSystem
     }
 
     @Override
-    public Catalog getLocationCatalog(String cursor, int pageSize, String unitSystem, String office,
+    public Catalog getLocationCatalog(String page, int pageSize, String unitSystem, String office,
                                       String idLike, String categoryLike, String groupLike) {
         //Now querying against AV_LOC2 as it gives us back the same information as querying against
         //location group views. This makes the code clearer and improves performance.
         //If there is a performance improvement by switching back to location groups and querying against
         //location codes (previous implementation used location_id) for joins, feel free to implement.
+        final AV_LOC2 avLoc2 = AV_LOC2.AV_LOC2;
         String locCursor = "*";
-        Condition condition = buildCatalogWhere(unitSystem, office, idLike, categoryLike, groupLike);
+        String searchOffice = office;
+        String curOffice = null;
+        Catalog.CatalogPage catPage = null;
         int total;
-        if (cursor == null || cursor.isEmpty()) {
+        Condition condition = avLoc2.LOCATION_ID.upper().likeRegex(idLike.toUpperCase())
+                              .and(avLoc2.LOCATION_CODE.notEqual(0L))
+                              .and(avLoc2.UNIT_SYSTEM.equalIgnoreCase(unitSystem));
+        if (page == null || page.isEmpty()) {
+            
+            if( categoryLike == null && groupLike == null) {
+                condition = condition.and(avLoc2.ALIASED_ITEM.isNull());
+            }
+            if (searchOffice != null) {
+                condition =
+                        condition.and(avLoc2.DB_OFFICE_ID.upper().eq(searchOffice.toUpperCase()));
+            }
 
+            if (categoryLike != null) {
+                condition.and(avLoc2.LOC_ALIAS_CATEGORY.upper().likeRegex(categoryLike.toUpperCase()));
+            }
+
+            if (groupLike != null) {
+                condition.and(avLoc2.LOC_ALIAS_GROUP.upper().likeRegex(groupLike.toUpperCase()));
+            }
             SelectConditionStep<Record1<Integer>> count = dsl.select(count(asterisk()))
-                .from(AV_LOC2.AV_LOC2)
+                .from(avLoc2)
                 .where(condition);
-
+            logger.info(count.getSQL(ParamType.INLINED));
```

Comment:
looks like debug output, lower to finer?

Reply:
yep.

Reply:
I've noticed a trend for info log levels, is it a pain to change those in a logging.properties file?

Reply:
no, I'm just very lazy... and forgetful apparently.